extensions/wikia/Search/classes/Config.php

```diff
@@ -720,25 +824,29 @@ public function getSkipBoostFunctions() {
 
 	/**
 	 * Allows us to abstract how we handle query service configuration
+	 *
 	 * @param string $service the query service without \\Wikia\\Search\\QueryService\\
 	 * @param bool $apply if set to false, we unset the queryservice if it's that value
+	 *
 	 * @throws \Exception
-	 * @return \Wikia\Search\Config
+	 * @return Config
 	 */
 	protected function setQueryService( $service, $apply ) {
-		if ( ! class_exists( '\\Wikia\\Search\\QueryService\\'.$service ) ) {
+		if ( !class_exists( '\\Wikia\\Search\\QueryService\\' . $service ) ) {
 			throw new \Exception( "Query service {$service} is not registered." );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS112)